contracts/svm/programs/express_relay/src/lib.rs

```diff
@@ -309,3 +463,94 @@ pub struct WithdrawFees<'info> {
 }
+
+#[derive(AnchorSerialize, AnchorDeserialize, Eq, PartialEq, Clone, Copy, Debug)]
+pub enum FeeToken {
+    Input,
+    Output,
+}
+
+/// For all swap instructions and contexts, input and output are defined with respect to the searcher
+/// So mint_input refers to the token that the searcher provides to the trader
+/// mint_output refers to the token that the searcher receives from the trader
+/// This choice is made to minimize confusion for the searchers, who are more likely to parse the program
+#[derive(AnchorSerialize, AnchorDeserialize, Eq, PartialEq, Clone, Copy, Debug)]
+pub struct SwapArgs {
+    pub amount_input:  u64,
+    pub amount_output: u64,
+    // The referral fee is specified in parts per million (e.g. 100 = 1 bp)
+    pub referral_fee:  u64,
+    // Token in which the fees will be paid
+    pub fee_token:     FeeToken,
+}
+
+#[derive(Accounts)]
+#[instruction(data: Box<SwapArgs>)]
+pub struct Swap<'info> {
+    /// Express relay configuration
+    #[account(seeds = [SEED_METADATA], bump)]
+    pub express_relay_metadata: Account<'info, ExpressRelayMetadata>,
+
+    /// Searcher is the party that sends the input token and receives the output token
+    pub searcher: Signer<'info>,
+
+    #[account(
+        mut,
+        token::mint = mint_input,
+        token::authority = searcher,
+        token::token_program = token_program_input
+    )]
+    pub searcher_input_ta: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(
+        mut,
+        token::mint = mint_output,
+        token::authority = searcher,
+        token::token_program = token_program_output
+    )]
+    pub searcher_output_ta: InterfaceAccount<'info, TokenAccount>,
+
+    /// Trader is the party that sends the output token and receives the input token
+    pub trader: Signer<'info>,
+
+    #[account(
+        mut,
+        token::mint = mint_input,
+        token::authority = trader,
+        token::token_program = token_program_input
+    )]
+    pub trader_input_ata: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(
+        mut,
+        token::mint = mint_output,
+        token::authority = trader,
+        token::token_program = token_program_output
+    )]
+    pub trader_output_ata: InterfaceAccount<'info, TokenAccount>,
+
+    // Accounts for the fee split
+    // TODO can we trust the Associated Token Program or should we check mint and token_program
+    /// CHECK: this is just used to check router_fee_receiver_ata is a valid ATA
+    pub router:                  UncheckedAccount<'info>,
+    pub router_fee_receiver_ata: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(mut,
+        token::authority = express_relay_metadata.relayer_signer,
+    )]
+    pub relayer_fee_receiver_ata: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(mut,
+        token::authority = express_relay_metadata.key(),
+    )]
+    pub protocol_fee_receiver_ata: InterfaceAccount<'info, TokenAccount>,
```

Comment:
`protocol` is potentially confusing, since we use `protocol` to mean the "router"/partner in other contexts. can we use a clearer term, e.g. `express_relay_fee_receiver_ata`?

Reply:
done